addons/mail/static/src/chatter/web/chatter_patch.js

```diff
@@ -261,6 +305,16 @@ patch(Chatter.prototype, {
         this.load(thread, ["suggestedRecipients"]);
     },
 
+    onUpdateProps(nextProps) {
+        if (
+            this.props.threadId !== nextProps.threadId ||
+            this.props.threadModel !== nextProps.threadModel
+        ) {
+            this.changeThread(nextProps.threadModel, nextProps.threadId, nextProps.webRecord);
+        }
+        super.onUpdateProps(nextProps);
```

Comment:
I see why you are doing this duplicate with onMount. The behaviour of `onMounted` and `onUpdateProps` should be exactly the same, so you can just refactoring into one and check the two props, in which one of it can be null if it is in `onMounted`